lib/domain/failures.dart

```diff
@@ -25,4 +25,26 @@ class Failure with _$Failure {
   const factory Failure.locationPermission() = _LocationPermission;
 
   const factory Failure.adapter() = _Adapter;
+
+  /// Failure occurring when the application's document directory could not
+  /// be located.
+  const factory Failure.documentDirectoryUnavailable() =
+      _DocumentDirectoryUnavailable;
+
+  /// Failure representing some databse error, consider using another failure
+  /// whenever possible.
+  const factory Failure.database() = _Database;
+
+  /// Failure occurring when the database could not be opened.
+  const factory Failure.databaseNotOpened() = _DatabaseNotOpened;
+
+  /// Failure occurring when the database could not be closed.
+  const factory Failure.databaseNotClosed() = _DatabaseNotClosed;
+
+  /// Failure occurring when something was not successfully saved to database.
+  const factory Failure.databaseNotSaved() = _DatabaseNotSaved;
```

Comment:
these are `domain` classes and should therefore not be aware of the underlying data source, whether it is a database, a simple file, hardcoded values should not matter.
Failures are more or less related to a use-cases, I think a more appropriate name would be `entitySave`, the given entity could not be saved or `entityDeletion` if the entity was not deleted.

I think `database`,  `databaseNotOpened` and  `databaseNoClosed` should not exist.
Same thing gos for the `adapter` failure actually and that is entirely my fault

Reply:
done